tasks/code_sync_status.rb

```diff
@@ -0,0 +1,93 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+# Puppet Task Name: code_sync_status
+require 'net/https'
+require 'uri'
+require 'json'
+require 'puppet'
+
+# CodeSyncStatus task class
+class CodeSyncStatus
+  def initialize(params)
+    @params = params
+  end
+
+  def execute!
+    puts syncstatus.to_json
+  end
+
+  def https
+    https = Net::HTTP.new('localhost', '8140')
+    https.use_ssl = true
+    https.cert = @cert ||= OpenSSL::X509::Certificate.new(File.read(Puppet.settings[:hostcert]))
+    https.key = @key ||= OpenSSL::PKey::RSA.new(File.read(Puppet.settings[:hostprivkey]))
+    https.verify_mode = OpenSSL::SSL::VERIFY_NONE
+    https
+  end
+
+  def apistatus
```

Comment:
By convention, method names begin with a lowercase letter. (Method names can begin with a capital letter, but that makes them look like constants.) When a method name is longer than one word, the usual convention is to separate the words with underscores `like_this` rather than using mixed case `likeThis`. [-- Source](https://www.oreilly.com/library/view/the-ruby-programming/9780596516178/ch06s02.html#:~:text=By%20convention%2C%20method%20names%20begin,than%20using%20mixed%20case%20likeThis%20.)

```suggestion
  def api_status
```